ServerCore/Pages/Threads/PuzzleThread.cshtml.cs

```diff
@@ -210,14 +210,27 @@ public async Task<IActionResult> OnPostAsync()
             m.ClaimerID = NewMessage.ClaimerID;
             m.PlayerID = NewMessage.PlayerID;
 
+            bool isMessageAdded = false;
             using (IDbContextTransaction transaction = _context.Database.BeginTransaction(System.Data.IsolationLevel.Serializable))
             {
-                _context.Messages.Add(m);
-                await _context.SaveChangesAsync();
-                transaction.Commit();
+                // Make sure to not add the message if already added previously (sometimes the user double clicks)
+                var newestMessage = _context.Messages
```

Comment:
we should await this if at all possible - I think LastAsync is a thing that can be used here.

Reply:
done!